File: frontend/src/types/finance.ts
export type AssetObservation = {
  date: string;
  close: number;
};

export type AssetContext = {
  ticker: string;
  name: string;
  latest_price: number | null;
  price_change: number | null;
  currency: string | null;
  market_cap: number | null;
  exchange: string | null;
  observations: AssetObservation[];
};

export type EdgarFiling = {
  form: string;
  filing_date: string;
  description: string;
  document_url: string | null;
};

export type CompanyContext = {
  company_name: string;
  cik: string;
  latest_filings: EdgarFiling[];
};

export type FinanceContext = {
  available: boolean;
  reason: string | null;
  market_id: number;
  market_title: string;
  asset_context: AssetContext | null;
  company_context: CompanyContext | null;
  context_ticker: string | null;
  context_kind: string | null;
};

export type EdgarBrowseResult = {
  available: boolean;
  reason: string | null;
  ticker: string;
  company_name: string;
  cik: string;
  filings: EdgarFiling[];
};

File: backend/app/schemas/finance.py
from datetime import date

from pydantic import BaseModel


class AssetObservationRead(BaseModel):
    date: date
    close: float


class AssetContextRead(BaseModel):
    ticker: str
    name: str
    latest_price: float | None = None
    price_change: float | None = None
    currency: str | None = None
    market_cap: int | None = None
    exchange: str | None = None
    observations: list[AssetObservationRead]


class EdgarFilingRead(BaseModel):
    form: str
    filing_date: date
    description: str
    document_url: str | None = None


class CompanyContextRead(BaseModel):
    company_name: str
    cik: str
    latest_filings: list[EdgarFilingRead]


class FinanceContextResponse(BaseModel):
    available: bool
    reason: str | None = None
    market_id: int
    market_title: str
    asset_context: AssetContextRead | None = None
    company_context: CompanyContextRead | None = None
    context_ticker: str | None = None
    context_kind: str | None = None


class EdgarBrowseResponse(BaseModel):
    available: bool
    reason: str | None = None
    ticker: str
    company_name: str
    cik: str
    filings: list[EdgarFilingRead]
